packages/hardhat/contracts/FundingVault.sol

```diff
@@ -118,6 +142,8 @@ contract FundingVault{
         developerFeeAddress =  _developerFeeAddress;
         developerFeePercentage = _developerFeePercentage;
         projectURL = _projectURL;
+        projectTitle = _projectTitle;
+        projectDescription = _projectDescription;
```

Comment:
For ERC20 token transfers, let's use OpenZeppelin's safeERC20 library's safeTransfer function. Then we don't need to check if the transfer was successful.

Reply:
There is no need to check if msg.value is zero. A zero amount will cause zero tokens to be transferred. ERC20 contracts do not allow 0 tokens to be transferred. Therefore, the transaction will revert anyway. This check is redundant.

Reply:
Fixed: Both FundingVault and Funding Vault Factory contract is using safetranser from OpenZepplin library

Reply:
> msg.value is zero
Including this one all all other redundant code instances where token value is checked for 0. is removed